OCPBUGS 20370:Added a note in supported ACME challenge types (#70742)
openshift/openshift-docs: modules/cert-manager-acme-challenges-types.adoc

+
For more information, see link:https://cert-manager.io/docs/configuration/acme/http01/[HTTP01] in the upstream cert-manager documentation.

> [NOTE]

Reviewer: Why the choice to make this a note rather than incorporating the text into the body of the module?

Author: It is supposed to be useful and additional information that users should know and expect while going through the description. And also, the requirement was initially to add a note too.

====

> HTTP-01 requires _Let's Encrypt_ servers to access the cluster's route. Note that using internal or private clusters behind the proxy to access, would fail the HTTP-01 validation checks required to issue a certificate.

Reviewer: "Let's Encrypt" is sometimes italicized and sometimes not within the docs set. You're just replicating what's already in this assembly, anyway, but I believe ISG guidance would be to not highlight the name because it is not difficult to distinguish from surrounding text.

Reviewer: I'm not sure if the subjects and objects are clear enough here. 

```suggestion
HTTP-01 requires that _Let's Encrypt_ servers can access the route of the cluster. If an internal or private cluster is behind a proxy, the HTTP-01 validations for certificate issuance fail.
```

Or something like that.

Author: Ack.

> HTTP-01 requires _Let's Encrypt_ staging environment to access the cluster's route to issue a certificate. Note that using internal or private clusters that are behind the proxy to access, would fail the HTTP-01 to issue a certificate. 

Reviewer 2: ```suggestion
HTTP-01 requires _Let's Encrypt_ servers to access the cluster's route. Note that using internal or private clusters behind the proxy to access, would fail the HTTP-01 validation checks required to issue a certificate. 
```

Author: Ack.

> The HTTP-01 challenge is restricted to port 80. For more information, see link:https://letsencrypt.org/docs/challenge-types/#http-01-challenge[Lets Encrypt].

Reviewer 2: To be more specific for readers, how about changing to:
```suggestion
The HTTP-01 challenge is restricted to port 80. For more information, see link:https://letsencrypt.org/docs/challenge-types/#http-01-challenge[HTTP-01 challenge] in the Let’s Encrypt documentation.
```

> The HTTP-01 challenge is restricted to port 80. For more information, see link:https://letsencrypt.org/docs/challenge-types/#http-01-challenge[Lets Encrypt].

Reviewer 3: ```suggestion
The HTTP-01 challenge is restricted to port 80. For more information, see link:https://letsencrypt.org/docs/challenge-types/#http-01-challenge[HTTP-01 challenge] (Let's Encrypt).
```
Is the current guidance for links out. Give the [page name](source).

> The HTTP-01 challenge is restricted to port 80 only. Allowing clients to specify arbitrary ports would make the challenge less secure, and therefore it is not allowed by the ACME standard.

Reviewer: ```suggestion
The HTTP-01 challenge is restricted to port 80. Allowing clients to specify arbitrary ports would make the challenge less secure, and therefore it is not allowed by the ACME standard.
```
Is the second sentence required?

Author: Specifying why it wouldn't work would be helpful to the users.

Author: That does make sense. Let me check with QE once if we can remove the sentence. Thanks for the review.

Author: @maxwelldb I can remove the sentence and add this link for reference- https://letsencrypt.org/docs/challenge-types/#http-01-challenge. It contains more information about Lets Encrypt. wdyt?

Reviewer: Totally up to you. I only parachuted in for peer review last week--I'm not aware of the context or customer needs of people who would be using this content.

Author: Thank you.
